dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/data/file/ImmediateFileWriter.kt

```diff
@@ -1,52 +1,44 @@
 /*
  * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
  * This product includes software developed at Datadog (https://www.datadoghq.com/).
- * Copyright 2016-2019 Datadog, Inc.
+ * Copyright 2016-2020 Datadog, Inc.
  */
 
 package com.datadog.android.core.internal.data.file
 
-import com.datadog.android.core.internal.data.DataMigrator
 import com.datadog.android.core.internal.data.Orchestrator
 import com.datadog.android.core.internal.data.Writer
 import com.datadog.android.core.internal.domain.Serializer
-import com.datadog.android.core.internal.threading.LazyHandlerThread
+import com.datadog.android.log.internal.utils.devLogger
 import com.datadog.android.log.internal.utils.sdkLogger
 import java.io.File
 import java.io.FileNotFoundException
 import java.io.FileOutputStream
 import java.io.IOException
 
-internal class FileWriter<T : Any>(
+internal class ImmediateFileWriter<T : Any>(
     private val fileOrchestrator: Orchestrator,
-    private val serializer: Serializer<T>,
-    private val dataMigrator: DataMigrator
-) : LazyHandlerThread(THREAD_NAME),
-    Writer<T> {
+    private val serializer: Serializer<T>
+) : Writer<T> {
 
-    init {
-        start()
-        post(Runnable {
-            dataMigrator.migrateData()
-        })
-    }
-
-    // region File Writer
+    // region Writer
 
     override fun write(model: T) {
-        post(Runnable {
-            val data = serializer.serialize(model)
+        val data = serializer.serialize(model)
 
-            if (data.length >= MAX_LOG_SIZE) {
-                // TODO RUMM-49 warn user that the log is too big !
-            } else {
-                synchronized(this) {
-                    writeData(data)
-                }
+        if (data.length >= MAX_ITEM_SIZE) {
```

Comment:
Just to make sure - are all crashlogs kept under this `MAX_ITEM_SIZE` 256Kb limit?

Reply:
256Kb is a really large size. In our crash message, the biggest payload is the stack trace that would usually be relatively small, and even in a stack overflow issue would still be under 256K